File: frontend/src/app/models/entity/Roles.ts
import { RolesUsuarios } from "./roles-usuario";

export class Roles {    
    constructor (
        public  servidor        ?: string,
	public  rolid           ?: number,
        public  hdgcodigo       ?: number,
	public  esacodigo       ?: number,
	public  cmecodigo       ?: number,
	public  codigorol       ?: string,
	public  nombrerol       ?: string,
        public  descripcionrol  ?: string,
        public  marcacheckgrilla?: boolean,
        public  rolesusuarios   ?: Array<RolesUsuarios>
    ) {  }
   
}

File: backend/Modulos/models/roles.go
package models

// Roles is...
type Roles struct {
	SERVIDOR       string `json:"servidor"`
	IDROL          int    `json:"rolid"`
	HDGCODIGO      int    `json:"hdgcodigo"`
	ESACODIGO      int    `json:"esacodigo"`
	CMECODIGO      int    `json:"cmecodigo"`
	CODIGOROL      string `json:"codigorol"`
	NOMBREROL      string `json:"nombrerol"`
	DESCRIPCIONROL string `json:"descripcionrol"`
	RolesUsuarios  []Roles `json:"rolesusuarios"`
}
